src/syndrome_decoder.jl

```diff
@@ -96,5 +98,89 @@ function syndrome_decode(pcm, pcmT, syndrome, max_iters, channel_probs, b2c, c2b
   end
 
   return Bool.(error), converged
+end
+
+# In-place variant
+# TODO: Currently very slow. Need to improve this
+function syndrome_decode!(decoder::BeliefPropagationDecoder, setup::BeliefPropagationSetup, syndrome::BitArray{1})
```

Comment:
Could you add a test that checks that this new code and your old code both produce the exact same result. I am uncomfortable that we basically have duplicated code here now. And could you add a todo comment somewhere about deleting the old function or implementing the old function in terms of the new function?

Reply:
Done